x/bank/legacy/v044/store.go

```diff
@@ -0,0 +1,51 @@
+package v044
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/address"
+	"github.com/cosmos/cosmos-sdk/x/bank/types"
+)
+
+// MigrateStore performs in-place store migrations from v0.43 to v0.44. The
+// migration includes:
+//
+// - Add an additional reverse index from denomination to address.
+func MigrateStore(ctx sdk.Context, storeKey sdk.StoreKey, cdc codec.BinaryCodec) error {
```

Comment:
@amaurym @aaronc can you quickly glance at this before I wrap up the unit tests?

I noticed the existing legacy migrations, e.g. v0.43, directly rely on the `types` package. I recall when initially implementing this model, we avoided direct dependency on types outside of `legacy` because they could change.

Reply:
> because they could change

I believe this assumption changed when we moved to Protobuf: since same-version protobuf types are additive and never breaking, we can use the same `types` folder for multiple migrations. When we'll bump proto versions, then yes, we'll change the old [output go pacakge](https://github.com/cosmos/cosmos-sdk/blob/d9fb4cf34d4ca65d2eca078dbd7edde084c884db/proto/cosmos/bank/v1beta1/bank.proto#L8) to point to a legacy folder.

(note: I initially followed exactly your model for migrations. But I ended up copy-pasting and cherry-picking a lot of stuff from `*.pb.go` into `legacy` folder, and it was a time-consuming process and error-prone. So I think using `types` in migrations is fine).

Reply:
otherwise this lgtm

Reply:
But the things in this case that reference types, aren't proto messages, but functions/messages. e.g. `CreateAccountBalancesPrefix` is a proto message.

Reply:
I'm happy to leave this as-is, I just wanted to raise awareness on it.

Reply:
ah yeah that's a good point, yeah I think copying those into the legacy folder is good practice, and only do `types.{some_proto_message}`

Reply:
so what are we copying into a legacy folder? functions that are legacy but not proto messages? i'm confused

Reply:
Functions and types that legacy logic depends on that are not proto-based. This is how legacy code always worked.